cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/persistence/aws/cloudsearch/CloudSearchEngine.java

```diff
@@ -145,25 +147,45 @@ public void update(final Collection<? extends Document> documents) {
             final DocumentConfiguration documentConfiguration = getDocumentConfiguration(documentClass);
             final String searchDomain = documentConfigurationHolder.schemaName() + "-"
                     + documentConfiguration.namespace();
-            final BatchDocumentUpdateRequest batchDocumentUpdateRequest = new BatchDocumentUpdateRequest(searchDomain);
+            for (final Collection<? extends Document> documentBatch : createDocumentBatches(documents)) {
```

Comment:
Although the batch logic is correct, won't the expression `createDocumentBatches()` be evaluated for each iteration of this loop? If it were moved out of the loop then it would only be called once which would be more efficient.